src/blob_gc_picker.cc

```diff
@@ -37,17 +44,42 @@ std::unique_ptr<BlobGC> BasicBlobGCPicker::PickBlobGC(
                      blob_file->file_number());
       continue;
     }
-    blob_files.push_back(blob_file.get());
 
-    batch_size += blob_file->file_size();
-    if (batch_size >= cf_options_.max_gc_batch_size) break;
+    if (!stop_picking) {
+      blob_files.push_back(blob_file.get());
+      batch_size += blob_file->file_size();
+      estimate_output_size +=
+          (blob_file->file_size() - blob_file->discardable_size());
+      if (batch_size >= cf_options_.max_gc_batch_size &&
+          estimate_output_size >= min_output_threshold) {
+        // stop pick file for this gc, but still check file for whether need
+        // trigger gc after this
+        stop_picking = true;
+      }
+    } else {
+      if (blob_file->GetDiscardableRatio() >=
```

Comment:
if not, break directly

Reply:
because the file is access in descending order so if I meet a file with discard_ratio less than threshold, then the discard ratio of file after this also less than threshold?

Reply:
on the restart, all the files' score is 1, but the discardable ratio is 0.